File: social-map/src/main/java/com/mapsocial/service/AdminService.java
package com.mapsocial.service;

import com.mapsocial.dto.request.admin.UpdateUserRequest;
import com.mapsocial.dto.response.admin.DashboardStatsResponse;
import com.mapsocial.dto.response.admin.UserManagementResponse;
import com.mapsocial.dto.response.shop.ShopResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface AdminService {
    DashboardStatsResponse getDashboardStats();

    Page<UserManagementResponse> getAllUsers(Pageable pageable, String search, Boolean includeDeleted);

    UserManagementResponse updateUser(UUID userId, UpdateUserRequest request);

    void softDeleteUser(UUID userId);

    void restoreUser(UUID userId);

    // Shop management
    Page<ShopResponse> getAllShopsAdmin(Pageable pageable, String search, Boolean includeDeleted);

    void softDeleteShop(UUID shopId);

    void restoreShop(UUID shopId);

    void softDeleteMultipleShops(List<UUID> shopIds);

    void updateShopStatus(UUID shopId, com.mapsocial.enums.ShopStatus status);
}

File: social-map-fe/src/services/adminService.js
import { api } from './apiClient';

// Lấy thống kê dashboard
export const getDashboardStats = async () => {
  return await api.get('/admin/dashboard/stats'); // api.get already returns response.data
};

// Quản lý users
export const getAllUsers = async (page = 0, size = 10, sortBy = 'createdAt', sortDirection = 'DESC', search = '', includeDeleted = false) => {
  const params = { page, size, sortBy, sortDirection };
  if (search) params.search = search;
  if (includeDeleted) params.includeDeleted = includeDeleted;
  return await api.get('/admin/users', { params });
};

export const updateUser = async (userId, data) => {
  return await api.put(`/admin/users/${userId}`, data);
};

export const deleteUser = async (userId) => {
  return await api.delete(`/admin/users/${userId}`);
};

export const restoreUser = async (userId) => {
  return await api.put(`/admin/users/${userId}/restore`);
};

export const updateUserRole = async (userId, role) => {
  return await api.put(`/admin/users/${userId}/role`, { role });
};

// Quản lý shops
export const getAllShopsAdmin = async () => {
  return await api.get('/admin/shops');
};

export const deleteShopAdmin = async (shopId) => {
  return await api.delete(`/admin/shops/${shopId}`);
};

export const updateShopStatus = async (shopId, status) => {
  return await api.put(`/admin/shops/${shopId}/status`, { status });
};

